sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsMeta.scala

```diff
@@ -170,6 +171,10 @@ abstract class RapidsMeta[INPUT <: BASE, BASE, OUTPUT <: BASE](
     }
   }
 
+  final def mustBeReplaced(because: String): Unit = {
+    mustBeReplacedReasons.get.add(because)
```

Comment:
The various mutable sets like `cannotBeReplacedReasons` and `mustBeReplaced` are initialized in `initReasons`:

```scala
def initReasons(): Unit = {
  cannotBeReplacedReasons = Some(mutable.Set[String]())
  ...
```
and I was basically copying the same pattern that we use in methods like `willNotWorkOnGpu`:

```scala
final def willNotWorkOnGpu(because: String): Unit = {
  cannotBeReplacedReasons.get.add(because)
  ...
```

Reply:
It might be cleaner to change these to `cannotBeReplacedReasons.foreach(_.add(because))` but it could also hide errors if we fail to initialize any of these sets.